src/sklearn_evaluation/models.py

```diff
@@ -0,0 +1,430 @@
+        self.report_section["guidelines"].append(guideline)
+
+    def get_dict(self) -> dict:
+        """
+        Return dict of the section
+        """
+        return self.report_section
+
+    def set_is_ok(self, is_ok):
+        """
+        Set if the reported test is valid
+        """
+        self.report_section["is_ok"] = is_ok
+
+    def set_include_in_report(self, include):
+        """
+        Set if should include this section in the report
+        """
+        self.report_section["include_in_report"] = include
+
+
+class ModelEvaluator(ModelHeuristics):
+    """
+    Generates model evaluation report
+    """
+
+    def __init__(self, model):
+        self.model = model
+        super().__init__()
+
+    def evaluate_balance(self, y_true):
+        """
+        Checks if model is balanced
+        """
+        balance_section = ReportSection("balance")
+
+        is_balanced = check_array_balance(y_true)
+
+        if is_balanced:
+            balance_section.set_is_ok(True)
+            balance_section.append_guideline("your model is balanced")
+        else:
+            balance_section.set_is_ok(False)
+            p = plot.target_analysis(y_true)
+            balance_section.append_guideline("Your test set is highly imbalanced")
+            balance_section.append_guideline(p)
+            balance_section.append_guideline(
+                "To tackle this, check out this "
+                "<a href='https://ploomber.io/blog/' target='_blank'>guide</a>"
+            )
+
+        self._add_section_to_report(balance_section)
+
+    def evaluate_accuracy(self, y_true, y_pred_test):
+        """
+        Measures how many labels the
+        model got right out of the total number of predictions
+        """
+        accuracy_threshold = 0.9
+        accuracy_section = ReportSection("accuracy")
+        accuracy = accuracy_score(y_true, y_pred_test)
+
+        balance = self.evaluation_state["balance"]
+
+        accuracy_section.append_guideline(f"Accuracy is {accuracy}")
+        if accuracy >= accuracy_threshold:
+            if balance["is_ok"]:
+                accuracy_section.append_guideline["is_ok"] = True
+                accuracy_section.append_guideline("You model is accurate")
+            else:
+                accuracy_section.set_is_ok(False)
+                accuracy_section.append_guideline(
+                    "Please note your model is unbalanced, "
+                    "so high accuracy could be misleading"
+                )
+
+        self._add_section_to_report(accuracy_section)
+
+    def evaluate_confusion_matrix(y_true, y_pred):
+        """
+        Checks how many of a classifier's predictions were correct,
+        and when incorrect, where the classifier got confused
+        """
+        # TODO: Implement
+        cm = confusion_matrix(y_true, y_pred)
+
+        # normalize
+        cm = cm.astype("float") / cm.sum(axis=1)[:, np.newaxis]
+
+        # TODO: Evaluate model by confusion matrix
```

Comment:
what's missing?

Reply:
I found an article on evaluating random forest model, they describe how to evaluate the model using the diagonal values. I didn't implement it since I'm not sure we need it.
https://medium.com/analytics-vidhya/evaluating-a-random-forest-model-9d165595ad56

I'll remove it from the code.